asf_tools/flood_map.py

```diff
@@ -69,14 +69,15 @@ def __call__(self, **kwargs):
             return tmax and tmin
 
     bounds = MyBounds()
-    x0 = [np.mean(water_levels)]
-    opt_res = optimize.basinhopping(_goal_ts, x0, niter=10000, niter_success=100, accept_test=bounds)
-    if opt_res.message[0] == 'success condition satisfied' \
-            or opt_res.message[0] == 'requested number of basinhopping iterations completed successfully':
-        best_water_level = opt_res.x[0]
-    else:
-        best_water_level = np.inf  # unstable solution.
-    return best_water_level
+    temp_wl = np.zeros(max(water_levels))
+    for i in range(1, max(water_levels)):
```

Comment:
This loop looks completely independent so I think you could use `multiprocessing`. 


Would need to put everything inside the for loop into a function, and then it could look like:

```python
from multiprocessing import Pool
from tqdm import tqdm

def fn(...):
    ...

iterations = list(range(1, max(water_levels))
with Pool() as p:
    temp_wl = list(
        tqdm(p.imap(fn, iterations), total=len(iterations))
    )
return np.nanmean(temp_wl)
```

That's probably the easiest way to speed things up intially (though it might need more memory then!)

Reply:
Great! I think that's a great idea. Is multiprocessing a standard package or should t also be added to enivronment.yml?

Reply:
I'm the standard lib! 😁

Reply:
Spectacular! Love this!